sources/packages/backend/libs/sims-db/src/entities/application-offering-change-requests.model.ts

```diff
@@ -0,0 +1,162 @@
+
+const REASON_MAX_LENGTH = 500;
+/**
+ * Represents the list of application specific offering change request,
+ * which is requested by institution.
+ */
+@Entity({ name: TableNames.ApplicationOfferingChangeRequests })
+export class ApplicationOfferingChangeRequest extends RecordDataModel {
+  /**
+   * Auto-generated sequential primary key column.
+   */
+
+  @PrimaryColumn({
+    name: "id",
+    nullable: false,
+  })
+  id: number;
+
+  /**
+   * Student Application id related to application specific offering change request.
+   */
+  @RelationId(
+    (applicationOfferingChangeRequest: ApplicationOfferingChangeRequest) =>
+      applicationOfferingChangeRequest.application,
+  )
+  applicationId: number;
+  /**
+   * Student Application on which the application specific offering change was requested.
+   */
+  @ManyToOne(() => Application, {
+    eager: false,
+    cascade: false,
+    nullable: false,
+  })
+  @JoinColumn({
+    name: "application_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  application: Application;
+
+  /**
+   * The new offering id created by the institution for the application specific offering change.
+   */
+  @RelationId(
+    (applicationOfferingChangeRequest: ApplicationOfferingChangeRequest) =>
+      applicationOfferingChangeRequest.requestedOffering,
+  )
+  requestedOfferingId: number;
+  /**
+   * The new or existing offering assigned by the institution for the application specific offering change.
+   */
+  @ManyToOne(() => EducationProgramOffering, {
+    eager: false,
+    cascade: false,
+    nullable: false,
+  })
+  @JoinColumn({
+    name: "requested_offering_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  requestedOffering: EducationProgramOffering;
+
+  /**
+   * The actual offering id that where requested for the application specific offering change by the institution.
+   */
+  @RelationId(
+    (applicationOfferingChangeRequest: ApplicationOfferingChangeRequest) =>
+      applicationOfferingChangeRequest.activeOffering,
+  )
+  activeOfferingId: number;
+  /**
+   *The actual offering that where requested for the application specific offering change by the institution.
+   */
+  @ManyToOne(() => EducationProgramOffering, {
+    eager: false,
+    cascade: false,
+    nullable: false,
+  })
+  @JoinColumn({
+    name: "active_offering_id",
+    referencedColumnName: ColumnNames.ID,
+  })
+  activeOffering: EducationProgramOffering;
+
+  /**
+   * Current status of application specific offering request change (e.g. In progress with the student,
+   * In progress with SABC, Approved, Declined by student, Declined by SABC).
+   */
+  @Column({
+    name: "application_offering_change_request_status",
+    type: "enum",
+    enum: ApplicationOfferingChangeRequestStatus,
+    enumName: "ApplicationOfferingChangeRequestStatus",
+  })
+  applicationOfferingChangeRequestsStatus: ApplicationOfferingChangeRequestStatus;
```

Comment:
Can we rename it to `applicationOfferingChangeRequetStatus` or `changeRequetStatus`?